### PR TITLE
Use Cluetrust GPX extension to add total distance

### PR DESCRIPTION
**Pull request by brianjmurrell**
**Describe the pull request**
To each trackpiont written to the GPX file.

**Link to the the issue**
Fixes: #1160

**License agreement**
By opening this pull request, you are providing your contribution under the _Apache License 2.0_ (see [LICENSE.md](LICENSE.md)).

### CODE REVIEW

File: src/main/java/de/dennisguse/opentracks/io/file/exporter/GPXTrackExporter.java [hunk 1]
```diff
@@ -312,7 +315,9 @@ public void writeCloseSegment() {
         printWriter.println("</trkseg>");
     }
 
-    public void writeTrackPoint(ZoneOffset zoneOffset, TrackPoint trackPoint, List<TrackPoint> sensorPoints) {
+    public Double writeTrackPoint(ZoneOffset zoneOffset, TrackPoint trackPoint, List<TrackPoint> sensorPoints, Double trackDistance) {
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/io/file/exporter/GPXTrackExporter.java:318]: Use Distance (specific value object; easier to understand than Double)

File: src/main/java/de/dennisguse/opentracks/io/file/exporter/GPXTrackExporter.java [hunk 2]
```diff
@@ -352,9 +357,10 @@ public void writeTrackPoint(ZoneOffset zoneOffset, TrackPoint trackPoint, List<T
                     trackPointExtensionContent += ("<opentracks:loss>" + ALTITUDE_FORMAT.format(cumulativeLoss) + "</opentracks:loss>\n");
                 }
 
-                Double cumulativeDistance = cumulateSensorData(trackPoint, sensorPoints, (tp) -> tp.hasSensorDistance() ? tp.getSensorDistance().toM() : null);
+                cumulativeDistance = cumulateSensorData(trackPoint, sensorPoints, (tp) -> tp.hasSensorDistance() ? tp.getSensorDistance().toM() : null);
                 if (cumulativeDistance != null) {
                     trackPointExtensionContent += ("<opentracks:distance>" + ALTITUDE_FORMAT.format(cumulativeDistance) + "</opentracks:distance>\n");
+                    trackPointExtensionContent += ("<cluetrust:distance>" + ALTITUDE_FORMAT.format(trackDistance + cumulativeDistance) + "</cluetrust:distance>\n");
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/io/file/exporter/GPXTrackExporter.java:363]: should be DISTANCE_FORMAT (was broken before :) )

**Review by dennisguse — APPROVED** [review]
Looks good.
Just nitpicking :)

**Comment by brianjmurrell** [comment]
Great feedback.  I will implement your suggestions before we land.  Thanks for taking the time.

**Comment by brianjmurrell** [comment]
I hope I have understood your suggestion to use `Distance` instead of `Double`.

As an aside, I note that `Distance` has interesting methods such as `plus`, `minus`, `of`, etc.  What is the advantage of that over implementing the more native `+` (`+=`), `-`, `==`, etc. operators?

**Comment by dennisguse** [comment]
Looks good; can you just squash everything into the first commit?

**Comment by brianjmurrell** [comment]
> Looks good; can you just squash everything into the first commit?

Done, and rebased on latest `main` for good measure.

**Comment by brianjmurrell** [comment]
Hrm.  Hold up on landing this...

**Comment by brianjmurrell** [comment]
Fixed.  Needed to use `.toM()` to get the `Double` value.

**Pull request merged**